pkg/webui/console/components/webhook-form/mapping.js

```diff
@@ -59,33 +58,86 @@ const mapHeadersTypeFormValueToWebhookHeadersType = formValue =>
-    },
-    webhook_id: values.webhook_id,
-  },
-  base_url: values.base_url,
-  format: values.format,
-  headers: mapHeadersTypeFormValueToWebhookHeadersType(values.headers),
-  downlink_api_key: values.downlink_api_key,
-  uplink_message: mapMessageTypeFormValueToWebhookMessageType(values.uplink_message),
-  join_accept: mapMessageTypeFormValueToWebhookMessageType(values.join_accept),
-  downlink_ack: mapMessageTypeFormValueToWebhookMessageType(values.downlink_ack),
-  downlink_nack: mapMessageTypeFormValueToWebhookMessageType(values.downlink_nack),
-  downlink_sent: mapMessageTypeFormValueToWebhookMessageType(values.downlink_sent),
-  downlink_failed: mapMessageTypeFormValueToWebhookMessageType(values.downlink_failed),
-  downlink_queued: mapMessageTypeFormValueToWebhookMessageType(values.downlink_queued),
-  downlink_queue_invalidated: mapMessageTypeFormValueToWebhookMessageType(
-    values.downlink_queue_invalidated,
-  ),
-  location_solved: mapMessageTypeFormValueToWebhookMessageType(values.location_solved),
-  service_data: mapMessageTypeFormValueToWebhookMessageType(values.service_data),
-})
+// Encode and decode basic auth header.
+
+export const decodeBasicAuthRequest = value => hasBasicAuth(value)
+
+export const encodeBasicAuthRequest = (value, fieldValue) => {
+  if (value) {
+    return [...(fieldValue || []), { key: 'Authorization', value: 'Basic ' }]
+  }
+
+  if (hasBasicAuth(fieldValue)) {
+    return fieldValue.filter(h => !isBasicAuth(h))
+  }
+
+  return fieldValue
+}
+
+export const decodeBasicAuthHeaderUsername = value => {
+  if (hasBasicAuth(value)) {
+    const encodedCredentials = getBasicAuthValue(value)
+    if (encodedCredentials) {
+      const decodedCredentials = atob(encodedCredentials)
+      const decodedUsername = decodedCredentials.slice(0, decodedCredentials.indexOf(':'))
+      return decodedUsername
+    }
+  }
+
+  return ''
+}
+
+export const mapCredentialsToAuthHeader = (forUsername, value, currentHeaders) => {
+  const basicAuthIndex = currentHeaders.findIndex(isBasicAuth)
+  if (basicAuthIndex !== -1) {
+    const encodedCredentials = getBasicAuthValue(currentHeaders)
+    if (encodedCredentials !== undefined) {
+      const decodedCredentials = atob(encodedCredentials)
+      const username = decodedCredentials.slice(0, decodedCredentials.indexOf(':'))
+      const password = decodedCredentials.slice(
+        decodedCredentials.indexOf(':') + 1,
+        decodedCredentials.length,
+      )
+      const newHeaders = [...currentHeaders]
+      if (forUsername && (value !== '' || password !== '')) {
+        newHeaders[basicAuthIndex].value = `Basic ${btoa(`${value}:${password}`)}`
+      } else if (!forUsername && (value !== '' || username !== '')) {
+        newHeaders[basicAuthIndex].value = `Basic ${btoa(`${username}:${value}`)}`
+      } else {
+        newHeaders[basicAuthIndex].value = 'Basic '
+      }
+      return newHeaders
+    }
+  }
+  return currentHeaders
```

Comment:
Nit pick (I think I introduced that): We usually add an empty line before `return` statements.